### PR TITLE
Visitor is actually an event listener

### PR DESCRIPTION
**Pull request by oldergod**
And I removed the validate method

### CODE REVIEW

File: wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/SchemaEncoderTest.kt [hunk 1]
```diff
@@ -230,10 +230,11 @@ class SchemaEncoderTest {
         .addMessageType(
           DescriptorProto.newBuilder()
             .setName("TestMessage")
-            .addEnumType(EnumDescriptorProto.newBuilder()
-              .setName("Nested")
-              .addValue(0, EnumValueDescriptorProto.newBuilder().setName("NESTED_UNDEFINED").setNumber(0))
-              .addValue(1, EnumValueDescriptorProto.newBuilder().setName("NESTED_DEFINED").setNumber(1))
+            .addEnumType(
```
Comment (oldergod) [inline wire-schema/src/jvmTest/kotlin/com/squareup/wire/schema/internal/SchemaEncoderTest.kt:233]: just formating

File: wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/ErrorCollector.kt [hunk 2]
```diff
@@ -20,16 +20,17 @@ package com.squareup.wire.schema
  * where they occurred within the schema.
  */
 class ErrorCollector {
-  private val errors: MutableList<String>
+  private val errorsBuilder: MutableList<String>
+  val errors: List<String> get() = errorsBuilder
```
Comment (swankjesse) [inline wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/ErrorCollector.kt:24]: this aught to be `errorsBuilder.toList()`

Comment (swankjesse) [inline wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/ErrorCollector.kt:24]:
```suggestion
  val errors: List<String> get() = errorsBuilder.toList()
```

File: wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/WireRun.kt [hunk 3]
```diff
@@ -280,19 +280,24 @@ class WireRun(
...
         }
       }
     }
-    visitors.forEach(Visitor::schemaHandlersEnd)
+    eventListeners.forEach(EventListener::schemaHandlersEnd)
 
-    errorCollector.throwIfNonEmpty()
+    try {
+      errorCollector.throwIfNonEmpty()
+    } catch (e: SchemaException) {
```
Comment (swankjesse) [inline wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/WireRun.kt:297]: Avoid exceptions for flow control here? As is this is ugly

```
val errorsSoFar = errorCollector.errors
if (errorsSoFar.isNotEmpty()) {
  eventListeners.forEach { it.runFailed(errorCollector.errors) }
  throw ...
}
```

**Review by swankjesse — APPROVED** [review]

File: wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/WireRun.kt (hunk 3 above)
Comment (oldergod) [reply wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/WireRun.kt:297]: fixed

File: wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/ErrorCollector.kt (hunk 2 above)
Comment (oldergod) [reply wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/ErrorCollector.kt:24]: fixed

**Pull request merged**